connection_test.go

```diff
@@ -34,3 +35,23 @@ func TestQueueDeclareOnAClosedConnectionFails(t *testing.T) {
 		log.Fatalf("queue.declare on a closed connection %s is expected to fail", conn)
 	}
 }
+
+func TestConcurrentClose(t *testing.T) {
+	conn, err := Dial("amqp://guest:guest@localhost:5672/")
+	if err != nil {
+		log.Fatalf("Could't connect to amqp server, err = %s", err)
+	}
+
+	wg := sync.WaitGroup{}
+	wg.Add(5)
```

Comment:
When I change this value to something higher than 5, I cannot get the suite to pass 20 times in a row, e.g.

```
for i in {1..20}; do go test -v -tags integration_test.go .\/...; done
```